tls/s2n_alerts.c

```diff
@@ -245,64 +228,25 @@ int s2n_process_alert_fragment(struct s2n_connection *conn)
     return 0;
 }
 
-int s2n_queue_writer_close_alert_warning(struct s2n_connection *conn)
+static S2N_RESULT s2n_queue_reader_alert(struct s2n_connection *conn, s2n_tls_alert_code code)
 {
-    POSIX_ENSURE_REF(conn);
-
-    uint8_t alert[2];
-    alert[0] = S2N_TLS_ALERT_LEVEL_WARNING;
-    alert[1] = S2N_TLS_ALERT_CLOSE_NOTIFY;
-
-    struct s2n_blob out = { 0 };
-    POSIX_GUARD(s2n_blob_init(&out, alert, sizeof(alert)));
-
-    /* If there is an alert pending, do nothing */
-    if (s2n_stuffer_data_available(&conn->writer_alert_out)) {
-        return S2N_SUCCESS;
-    }
-
-    if (!s2n_alerts_supported(conn)) {
-        return S2N_SUCCESS;
-    }
-
-    POSIX_GUARD(s2n_stuffer_write(&conn->writer_alert_out, &out));
-
-    return S2N_SUCCESS;
-}
-
-static int s2n_queue_reader_alert(struct s2n_connection *conn, uint8_t level, uint8_t error_code)
-{
-    POSIX_ENSURE_REF(conn);
-
-    uint8_t alert[2];
-    alert[0] = level;
-    alert[1] = error_code;
-
-    struct s2n_blob out = { 0 };
-    POSIX_GUARD(s2n_blob_init(&out, alert, sizeof(alert)));
-
-    /* If there is an alert pending, do nothing */
-    if (s2n_stuffer_data_available(&conn->reader_alert_out)) {
-        return S2N_SUCCESS;
-    }
-
-    if (!s2n_alerts_supported(conn)) {
-        return S2N_SUCCESS;
+    RESULT_ENSURE_REF(conn);
+    if (!conn->reader_alert_out) {
```

Comment:
Nit
```suggestion
    if (conn->reader_alert_out == 0) {
```
reader_alert_out isn't a bool so this is more clear to me. But maybe that's just me XD

Reply:
I disagree, but I think this is a style thing. Technically in the C standard, "if" doesn't even 'operate' on `bool`, it just checks if the expression given is equal to 0. `conn->reader_alert_out == 0` technically evaluates to `1` or `0` too, not `true` or `false` ;) I always read `!x` as "x is not set" (thinking about pointers), which I think reflects my intention here better.